modules/st2-filter/directive.js

```diff
@@ -1,31 +1,85 @@
 'use strict';
 
 angular.module('main')
-  .directive('st2Filter', function () {
+  .directive('st2Filter', function ($parse) {
 
     return {
       restrict: 'C',
       scope: {
-        type: '@',
-        label: '@',
-        filters: '='
+        defaultLabel: '@label',
+        multiple: '@',
+        items: '=',
+        activeItemsSrc: '=activeItems'
       },
       templateUrl: 'modules/st2-filter/template.html',
-      link: function postLink(scope, element) {
+      link: function postLink(scope, element, attrs) {
+        var onChange = $parse(attrs.onChange);
+        var listElement = element[0].getElementsByClassName('st2-filter__list')[0];
+        var searchBarElement = element[0].getElementsByClassName('st2-filter__search-bar')[0];
+
+        scope.refreshItemsToShow = function () {
+          var activeItems = scope.activeItems.filter(function (item) {
+            return !scope.search || item.indexOf(scope.search) !== -1;
+          });
+          var otherItems = scope.items.filter(function (item) {
+            return scope.activeItems.indexOf(item) === -1 &&
+                   (!scope.search || item.indexOf(scope.search) !== -1);
+          }).sort();
+
+          scope.itemsToShow = activeItems.concat(otherItems);
+        };
+
         scope.toggle = function () {
+          listElement.scrollTop = 0;
+          scope.search = '';
+          scope.refreshItemsToShow();
           element.toggleClass('st2-filter--active');
+          searchBarElement.style.minWidth = searchBarElement.clientWidth + 'px';
+          searchBarElement.focus();
         };
 
         scope.pick = function (name) {
-          var o = _.clone(scope.$root.state.params);
```

Comment:
This stuff didn't belong here, so I moved it to the `updateFiltersAndGo` method in the root scope. Not sure if it's the best solution, though.